lib/go/thrift/framed_transport.go

```diff
@@ -155,7 +189,7 @@ func (p *TFramedTransport) readFrame() error {
 		return err
 	}
 	size := binary.BigEndian.Uint32(buf)
-	if size < 0 || size > p.maxLength {
+	if size < 0 || size > uint32(p.cfg.GetMaxFrameSize()) {
```

Comment:
The code was there before but I really wonder when size will become negative on an uint32?

Reply:
Before `p.maxLength` actually has the type of `uint32` instead of `int32`.

To become negative when converting to `int32` the `uint32` value needs to be at least 2GB (MaxInt32 is `1<<31-1`). I'm not sure how much do we really care about that use case. If we do care about it I can change the types of `MaxMessageSize` and `MaxFrameSize` both from `int32` to `uint32`.

Reply:
I very much doubt this is a legitimate possibility, `int32` seems fine to me.